snyk/client.py

```diff
@@ -40,13 +42,18 @@ def __init__(
             "Authorization": "token %s" % self.api_token,
             "User-Agent": user_agent,
         }
-        self.api_post_headers = self.api_headers
+        self.api_post_headers = copy.deepcopy(self.api_headers)
```

Comment:
```suggestion
        self.api_post_headers = dict(self.api_headers)
```
Why do we need to copy them?

Reply:
Since both api post headers and patch post headers have different content-type, I wanted a way to avoid side-effects(since simple assigning creates a shallow copy and dictionaries are mutable in python) when changing content-type. The solution given by you might be better rather than creating a deep-copy.

Reply:
So my suggestion is to set those headers explicitly in the specific function.
Otherwise it's confusing.

Reply:
This might introduce duplicate code, since both V1 and REST API uses the same headers.

Reply:
Again, it's a suggestion only.  In my opinion it is better to duplicate code and make them explicit rather than start looking at who modified what header.  Moreover, if you do that you will be able to get rid of the `api_post_headers` property.